angular/lib/src/core/linker/app_view.dart

```diff
@@ -654,10 +661,31 @@ abstract class AppView<T> {
   // parameter of the event listener is a subclass of Event. The Event passed in
   // from EventTarget.addEventListener() can then be safely coerced back to its
   // known type.
+
+  /// Used to bind to event streams where the event is passed to the component.
+  ///
+  /// Returns a function that invokes a one-argument event [handler].
   void Function(E) eventHandler1<E, F extends E>(void Function(F) handler) {
+    if (isDevMode) {
+      return _eventHandler1Debug<E, F>(handler);
+    }
     return (E event) {
       markPathToRootAsCheckOnce();
-      appViewUtils.eventManager.getZone().runGuarded(() => handler(event as F));
+      _ngZone.runGuarded(() {
+        handler(event as F);
+      });
+    };
+  }
+
+  /// Variant of [eventHandler1] that provides better runtime errors.
+  void Function(E) _eventHandler1Debug<E, F extends E>(void Function(F) fn) {
```

Comment:
So we could just put this in `DebugAppView.eventHandler1()` instead. Is this future-proofing for when we remove all the debug-related code?

Reply:
Yes. DebugAppView already does not have event handlers.

Reply:
Oh yeah it's not as simple as wrapping the call to `super`, since the check has to happen inside the returned closure, so you'd need to fully rewrite the code in the debug view anyways, which would probably be more prone to error and getting out of sync. This approach is better.